src/cli-flags.ts

```diff
@@ -1,8 +1,21 @@
-// Note: Consider using for example yargs if adding more flags (especially
-//       ones that should contain values of type other than boolean).
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
 
-type CliFlag = "--useMirroredNetworking";
+const argv = yargs(hideBin(process.argv))
+  .option("useMirroredNetworking", {
+    type: "boolean",
+    description: "Use mirrored networking",
+  })
+  .option("host", {
+    type: "string",
+    description: "Manually specify the host address",
+  })
+  .help().argv as { [key: string]: any }; // Explicitly typing argv
```

Comment:
use `satisfies` keyword to not lose the typescript definition for arguments

```suggestion
  .help().argv satisfies { [key: string]: any };
```

Reply:
Hi @lucasecdb, Thanks for the suggestion!

Unfortunately this breaks on line 16, I believe because of a problem with the library type definition:

```ts
Diagnostics:
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ [x: number]: string; [x: string]: unknown; length: number; toString: (() => string) & (() => string); toLocaleString: (() => string) & (() => string); pop: () => string | undefined; push: (...items: string[]) => number; ... 33 more ...; $0: string; } | Promise<...>'.
  No index signature with a parameter of type 'string' was found on type '{ [x: number]: string; [x: string]: unknown; length: number; toString: (() => string) & (() => string); toLocaleString: (() => string) & (() => string); pop: () => string | undefined; push: (...items: string[]) => number; ... 33 more ...; $0: string; } | Promise<...>'. [7053]
```

Reply:
this is because of the promise, you just need to add `await` to line 4